prov/efa/src/efa_hmem.c

```diff
@@ -123,12 +123,14 @@ static int efa_domain_hmem_info_init_cuda(struct efa_domain *efa_domain)
 		return 0;
 	}
 
-	cuda_ret = ofi_cudaMalloc(&ptr, len);
-	if (cuda_ret != cudaSuccess) {
-		EFA_WARN(FI_LOG_DOMAIN,
-			 "Failed to allocate CUDA buffer: %s\n",
-			 ofi_cudaGetErrorString(cuda_ret));
-		return 0;
+	if (!ofi_hmem_p2p_disabled()) {
+		cuda_ret = ofi_cudaMalloc(&ptr, len);
+		if (cuda_ret != cudaSuccess) {
+			EFA_WARN(FI_LOG_DOMAIN,
+				"Failed to allocate CUDA buffer: %s\n",
+				ofi_cudaGetErrorString(cuda_ret));
+			return 0;
+		}
```

Comment:
If p2p is disabled, shouldn't you skip all the subsequent ibv reg as well?

Reply:
Do we just set info->p2p_supported_by_device to false if p2p is disabled? I see https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_hmem.c#L36-L37 fall back to FI_HMEM_SYSTEM initialization logic when p2p is unavailable

Reply:
At the minimum, we should fix the obvious behavior on L137 - in this case the user explicitly disabled p2p should we should respect that.

Overall I suggest we audit CUDA/Neuron/Synapseai as a whole - if p2p is disabled by the user, we should skip the p2p checks entirely. CUDA is not special in this case.